dradis/dradis-ce
# Auto Save Whodunnit

### Summary

At the point we use a background job we no longer have access to PaperTrail.request.whodunnit because we're not operating on a different thread. To correctly apply who did it, we need to pass in the param when the job is serialized.

### How to test

- Log in with two tabs, and two different users
- Edit a resource with user A
- Trigger auto save by waiting 3 minutes, or navigating to a different page
- With User B validate the the auto-save occured
- With User B edit the resource, make no changes and press "Update"
- Verify on the revision history user B is credited with the update

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List
